lib/ansible/modules/cloud/azure/azure_rm_servers_facts.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2017 Zim Kalinowski, <zikalino@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_servers
+version_added: "2.5"
+short_description: Get Servers facts.
+description:
+    - Get facts of Servers.
+
+options:
+    resource_group_name:
+        description:
+            - The name of the resource group that contains the resource. You can obtain this value from the Azure Resource Manager API or the portal.
+        required: True
+    server_name:
+        description:
+            - The name of the server.
+        required: False
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Zim Kalinowski (@zikalino)"
+
+'''
+
+EXAMPLES = '''
+      - name: Get instance of Servers
+        azure_rm_servers:
+          resource_group_name: "{{ resource_group_name }}"
+          server_name: "{{ server_name }}"
+
+      - name: List instances of Servers
+        azure_rm_servers:
+          resource_group_name: "{{ resource_group_name }}"
+'''
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from msrestazure.azure_operation import AzureOperationPoller
+    from azure.mgmt.sql import SqlManagementClient
+    from msrest.serialization import Model
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMServersFacts(AzureRMModuleBase):
+    def __init__(self):
+        # define user inputs into argument
+        self.module_arg_spec = dict(
```

Comment:
resource_group, name

Reply:
Can support find list by tags?

Reply:
not sure, there's no such function available

Reply:
It do the filter locally after get the resources

Reply:
[https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/azure/azure_rm_availabilityset_facts.py#L140][]